script.js

```diff
@@ -167,3 +205,172 @@ function subButtonPressed() {
     alert("Złożono zamówienie");
     refreshCart();
 }
+
+function clearCartButtonPressed() {
+    cart = [];
+    cost = 0;
+    document.getElementById("cost").innerHTML = "Głodny? Zamów naszą pizzę";
+    cartLen = 0;
+    refreshCart();
+    deleteSubButton();
+    deleteClearCartButton();
+}
+
+
+function readCartFromLocalStorage() {
+    return JSON.parse(localStorage.getItem("cart"));
+}
+
+
+function saveCartToLocalStorage() {
+    localStorage.setItem("cart", JSON.stringify(cart));
+}
+
+
+function sortPizzasList(how) {
+    let id = [];
+    let title = [];
+    let price = [];
+    let image = [];
+    let ingredients = [];
+    let helper = [];
+
+    for (let i = 0; i < obj.length; i++) {
+        id[i] = obj[i].id;
+        title[i] = obj[i].title;
+        price[i] = obj[i].price;
+        image[i] = obj[i].image;
+        ingredients[i] = obj[i].ingredients;
+    }
+
+    switch (how) {
+        case "AZ":
+            title.sort();
+            break;
+        case "ZA":
+            title.sort();
+            title.reverse();
+            break;
+        case "09":
+            price.sort(function (a, b) {
+                return a - b
+            });
+            break;
+        case "90":
+            price.sort(function (a, b) {
+                return a - b
+            });
+            price.reverse();
+            break;
+    }
+
+    let ind = [];
+    let j = 0;
+    let index = 0;
+
+    if (how == "AZ" || how == "ZA") {
+        for (let i = 0; i < obj.length; i++) {
+            ind = findIndex(title[i], "title");
+            index = ind[j];
+            j++
+            if (j >= ind.length) {
+                j = 0;
+            }
+            helper.push(
+                {
+                    id: id[index],
+                    title: title[i],
+                    price: price[index],
+                    image: image[index],
+                    ingredients: ingredients[index]
+                });
+        }
+    } else if (how == "09" || how == "90") {
+        for (let i = 0; i < obj.length; i++) {
+            ind = findIndex(price[i], "price");
+            index = ind[j];
+            j++
+            if (j >= ind.length) {
+                j = 0;
+            }
+
+            helper.push(
+                {
+                    id: id[index],
+                    title: title[index],
+                    price: price[i],
+                    image: image[index],
+                    ingredients: ingredients[index]
+                });
+        }
+    }
+    obj = helper;
+}
```

Comment:
Musisz swoją funkcje sortowania podpiąć na `change` event `<select>`-a , ie. 
```
const selectElement = document.querySelector('#sort-list')
selectElement.addEventListener('change', sortPizzasList)
```

```
function sortPizzasList (event, list) {
  switch (event.target.value) {
        case "AZ":
            list.sort();
            break;
        case "ZA":
            list.sort();
            list.reverse();
            break;
        case "09":
            list.sort(function (a, b) {
                return a - b
            });
            break;
        case "90":
            list.sort(function (a, b) {
                return a - b
            });
            price.reverse();
            break;
    }
}
```

Reply:
Nie udało mi się zaimplementować tego w moim kodzie, dlatego użyłem osobnych funkcji sortujących

Reply:
Co do komentarza dotyczącego localStorage, nie udało mi się odtworzyć tego błędu, dlatego jeśli wciąż występuje to nie byłem w stanie go poprawić